### DISCUSSION TITLE
WorkloadSpread 可以增加对流量的管理吗

### DISCUSSION THREAD

**Discussion by FireSoilD** [openkruise/kruise#1550] [Ideas]
我们使用ws来管理工作负载，当触发扩容时会将pod分布到eci节点，此时希望eci节点的pod拥有更高的调度权重，或者可以自定义调度权重

**Comment by furykerry** [comment]
workloadspread 可以给分配到的subset配置需要做的patch， 如果流量管理可以读取pod上的数据（例如metadata），应该就可以干预流量调度的权重。 不过， 为啥需要给分配到ECI 的pod，更多的流量权重呢？

**Reply by FireSoilD** [reply]
为应对流量，扩容出来的节点规格比较好，而且是独享资源，希望能承载更多的流量

**Reply by FireSoilD** [reply]
补充：
此前，流量高峰期也遇到过ecs节点超时较多的情况
我们会给eci的subset打上 `set: eci` 标签，区别于ecs的subset的 `set: ecs` 标签

**Reply by FireSoilD** [reply]
我们此前也使用 ws 为不同的 subset patch 不同版本的 image，实现基于 pod 副本权重的灰度，希望 ws 能在流量管理这块更进一步，提供更多细粒度的管控

**Reply by furykerry** [reply]
给pod打不同pod的化，是可以给eci和ecs 的pod 创建不同的service。 一种办法是通过ingress 之类的gateway 来调整不同service的权重， 不过这个需要上层去调整。

**Reply by FireSoilD** [reply]
我看 rollouts 主要用于在交付期间的流量管理，好像还没有上述需求的解决方案

**Reply by furykerry** [reply]
你弹eci的时候能指定规格啥的， 保证和ecs的性能一致么， 这种场景如果还要考虑流量调度感觉不必要的复杂了

**Reply by FireSoilD** [reply]
扩容会创建规格更高的实例，此时Ingress后端存在运行于 ecs 和 eci 两类节点的 pods，并且 ecs 是混部，所以希望 eci 的 pod 的调度权重更高

**Reply by FireSoilD** [reply]
需要根据 特定标签 的 pods 的健康状态动态调整流量权重，确实有些复杂
